azuread/table_azuread_application.go

```diff
@@ -0,0 +1,119 @@
+			{Name: "identifier_uris", Type: proto.ColumnType_JSON, Description: "The URIs that identify the application within its Azure AD tenant, or within a verified custom domain if the application is multi-tenant. "},
+			{Name: "info", Type: proto.ColumnType_JSON, Description: "Basic profile information of the application such as app's marketing, support, terms of service and privacy statement URLs. The terms of service and privacy statement are surfaced to users through the user consent experience."},
+			{Name: "key_credentials", Type: proto.ColumnType_JSON, Description: "The collection of key credentials associated with the application."},
+			{Name: "owner_ids", Type: proto.ColumnType_JSON, Hydrate: getApplicationOwners, Transform: transform.FromValue(), Description: "Id of the owners of the application. The owners are a set of non-admin users who are allowed to modify this object."},
+			{Name: "parental_control_settings", Type: proto.ColumnType_JSON, Description: "Specifies parental control settings for an application."},
+			{Name: "password_credentials", Type: proto.ColumnType_JSON, Description: "The collection of password credentials associated with the application."},
+			{Name: "spa", Type: proto.ColumnType_JSON, Description: "Specifies settings for a single-page application, including sign out URLs and redirect URIs for authorization codes and access tokens."},
+			{Name: "tags_src", Type: proto.ColumnType_JSON, Description: "Custom strings that can be used to categorize and identify the application.", Transform: transform.FromField("Tags"),},
+			{Name: "web", Type: proto.ColumnType_JSON, Description: "Specifies settings for a web application."},
+
+			// // Standard columns
+			{Name: "tags", Type: proto.ColumnType_JSON, Description: ColumnDescriptionTags, Transform: transform.From(applicationTags)},
+			{Name: "title", Type: proto.ColumnType_STRING, Description: ColumnDescriptionTitle, Transform: transform.FromField("DisplayName", "ID")},
+			{Name: "tenant_id", Type: proto.ColumnType_STRING, Description: ColumnDescriptionTenant, Hydrate: plugin.HydrateFunc(getTenantId).WithCache(), Transform: transform.FromValue()},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listAdApplications(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	session, err := GetNewSession(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	client := msgraph.NewApplicationsClient(session.TenantID)
+	client.BaseClient.Authorizer = session.Authorizer
+
+	// TODO filters
+	input := odata.Query{}
+	filter := ""
+	input.Filter = filter
+
+	applications, _, err := client.List(ctx, input)
+	if err != nil {
+		if isNotFoundError(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	for _, application := range *applications {
+		d.StreamListItem(ctx, application)
+	}
+	return nil, err
+}
+
+// Hydrate Functions
+
+// we didn't add the get function as it retries 5 times on 404 errors
```

Comment:
What get function is this referring to? Are we missing any information if we don't have the get function?